### ISSUE TITLE
Definition of DFLT script is vague

### ISSUE THREAD

**Issue by davelab6** [MicrosoftDocs/typography-issues#558]
As @tiroj said at 
https://twitter.com/TiroTypeworks/status/1297013154682040323?s=19 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1cb42967-20ca-9270-4572-65315127fcc5
* Version Independent ID: 1fa413f1-71cf-ac2a-0999-70dd5d8ac4cd
* Content: [OpenType layout common table formats - Typography](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2)
* Content Source: [typographydocs/opentype/spec/chapter2.md](https://github.com/MicrosoftDocs/typography/blob/live/typographydocs/opentype/spec/chapter2.md)
* Product: **typography**
* GitHub Login: @PeterCon
* Microsoft Alias: **PeterCon**

**Comment by tiroj** [comment]
In case my Twitter comments were also gloriously vague, in the manner of that medium:

My recollection of the specification of the DFLT script tag to OpenType is that it occurred in one of the early updates, and was not something that had been part of the original TrueType Open model that formed the basis of OTL. As I recall, Adobe proposed the DFLT script tag, but Microsoft disagreed that there was a need for it. I think the wording in the spec reflects this: it states the basis under which Microsoft was willing to accept DFLT into the spec, but really doesn't seem to be encouraging its use: you may do this, but we don't expect it.

What is lacking in the specification is 
a) examples of use of DFLT to help font makers understand when it might be useful;
b) clarity about when DFLT shaping would be applied and how.

Based on recent discussion about DFLT on Twitter and in the [noto-fonts repo](https://github.com/googlefonts/noto-fonts/issues/576) it seems that at least some shaping paths would permit DFLT to be used for any script that does not require specific complex shaping requirements (such as would require the script to be passed to a specific script engine or to USE). That implies the existence of something like a Universal SIMPLE Shaping Engine, which I understand HarfBuzz to include—and perhaps other OTL implementations?—, but meanwhile Andrew Glass has confirmed my own inclination that any newly encoded script could be passed to USE, regardless of simplicity/complexity, and that suggests to me that DFLT could also be passed to USE.

**Comment by PeterCon** [comment]
I might be able to think of some clarification of "DFLT". But it seems to me that intended meaning and usage should really be worked out in the context of improving / creating a family of shaping specs.

**Comment by tiroj** [comment]
I agree. The DFLT script tag is one of those places where the font format strays towards an implementation format, but doesn’t quite get there. So a surprising amount of text is dedicated to describing the possibility of the DFLT script tag without really saying when it should be used or how it should be processed.

**Comment by behdad** [comment]
Please don't write `dflt` if you want to refer to `DFLT`!  There's more than enough confusion around that:

In OpenType, there's no tag for "default *language*". The file format has a `DefaultLangSys` for that. But it is my understanding that in Volt, one uses a tag of `dflt` to specify rules for the `DefaultLangSys`. I believe the `dflt` tag had even leaked into the OpenType spec at some point and caused a lot of confusion. People still refer to the `dflt` LangSys tag...

Now to `DFLT` script tag: In HarfBuzz if we don't find the any of the OT Layout script tags for the specific script of the text in the font, we try with `DFLT`, and if that is not found either, we try `dflt`, because, let me quote HarfBuzz source code:
```
  /* try with 'dflt'; MS site has had typos and many fonts use it now :(.
   * including many versions of DejaVu Sans Mono! */
```
and after that we try `latn`:
```
  /* try with 'latn'; some old fonts put their features there even though
     they're really trying to support Thai, for example :( */
```
because reports suggested that Uniscribe finds those features...

**Comment by NorbertLindenberg** [comment]
That leak in Volt has flooded the shaping engine descriptions – in every one of them it’s “strongly recommended to include the ‘dflt’ language tag in all OpenType fonts”.

**Comment by PeterCon** [comment]
Norbert: valid critique wrt 'dflt'. Of course, most of those specs were written at a time when VOLT was the only tool available. But that's not an excuse for today.

**Comment by tiroj** [comment]
dflt is also used for DefaultLangSys in AFDKO .fea syntax. It’s one of those things where the interaction of people making fonts with the format is almost universally understood in terms of a higher level input language. Not the worst example of such: I meet people who think languages systems are a property of feature lookups, because that's how .fea codes them.
